dotnetv3/Bedrock-runtime/Tests/ActionTest_Base.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved. 
+// SPDX-License-Identifier: Apache-2.0
+
+using System.Diagnostics;
+
+namespace BedrockRuntimeTests
+{
+    public abstract class ActionTest_Base
+    {
+        protected string _projectRoot;
+        private readonly ProcessStartInfo _processStartInfo;
+
+        protected ActionTest_Base()
+        {
+            _projectRoot = GetProjectRoot();
+            _processStartInfo = new ProcessStartInfo
+            {
+                RedirectStandardOutput = true,
+                UseShellExecute = false,
+                CreateNoWindow = true
+            };
```

Comment:
Is there a way we can run the code by filename instead of generically assembling  a path and running it? Could you wrap it in a class and method?

Reply:
I'm not sure I understand what you mean. Can you elaborate?

Reply:
Wrap the example code in a class and method? I'd rather keep it as a top-level script to keep it as clutter-free and readable as possible. Class and method signatures don't add anything except additional lines of code, especially in C#. 

Also: Wrapping the code in class and method adds 2 indentations (8 leading whitespaces) to every important line of code. The examples are already hard to read without any leading whitespaces, especially on mobile (see attached screenshots from Safari on the iPhone). We should make sure to use every bit of screen real-estate.

[image](https://github.com/awsdocs/aws-doc-sdk-examples/assets/604796/94310732-ae99-4b2b-bf74-11ac440a2125)
![image](https://github.com/awsdocs/aws-doc-sdk-examples/assets/604796/a348b7c9-ad6a-4f03-bc9d-b07a82e71efc)

Reply:
Well, I am not convinced that Safari on an iPhone is popular for customers to view C# code examples. But I hear your point about indents. Top level statements do help, but they should also be testable. Using a class and method is one way, have you looked into any other methods to test the top level statements where they can be referenced by name instead of a generic script execution? Possibly by using the `InternalsVisibleTo` property in the project?

Reply:
InternalsVisibleAt probably won't help me here, but I will have a look at the CodeAnalysis.Scripting packages. With them I should be able to ensure that the tests can't execute arbitrary non-csx files.

Reply:
I have replaced the low-level execution of the executables with the below, ensuring that only actual CSX scripts can be executed:

```
Microsoft.CodeAnalysis.CSharp.Scripting.CSharpScript.RunAsync(csxScript)
```

Reply:
Ok, that is improved.